File: TCDNew-master/src/ui.wince/Cen.Wms.Client/Actions/Data/Auth/UserLogin.cs
using System;
using System.Linq;
using System.Net;
using Cen.Wms.Client.Actions.UI.Utility;
using Cen.Wms.Client.Common;
using Cen.Wms.Client.Models.Dtos;
using Cen.Wms.Client.Models.Exceptions;
using Cen.Wms.Client.Services;
using NLog;

namespace Cen.Wms.Client.Actions.Data.Auth
{
    class UserLogin
    {
        public static UserTokenResp Run(UserCredentials userCredentials)
        {
            UserTokenResp result = null;

            try
            {
                result = RpcService.Instance.UserLogin(userCredentials);
            }
            catch (RpcException exception)
            {
                ShowModalMessage.Run(Messages.ErrorServer, string.Join(Environment.NewLine, exception.Errors.Select(e => e.ErrorText).ToArray()));
            }
            catch (WebException exception)
            {
                var logger = LogManager.GetLogger(Messages.LoggerNetName);
                logger.Error(exception);

                ShowModalMessage.Run(Messages.ErrorTransport, Messages.ErrorUserLogin);
            }
            catch (Exception exception)
            {
                var logger = LogManager.GetLogger(Messages.LoggerLocalName);
                logger.Error(exception);

                ShowModalMessage.Run(Messages.ErrorUnknown, Messages.ErrorUserLogin);
            }

            return result;
        }
    }
}

File: TCDNew-master/src/ui.android/lib/actions/data/auth/UserLogin.dart
import 'dart:async';

import 'package:flutter/material.dart';
import 'package:tiq/actions/ui/utility/ShowModalMessage.dart';
import 'package:tiq/models/dtos/UserCredentials.dart';
import 'package:tiq/models/dtos/UserTokenResp.dart';
import 'package:tiq/models/exceptions/RpcException.dart';
import 'package:tiq/common/Messages.dart';
import 'package:tiq/services/RpcService.dart';

class UserLogin {
  Future<UserTokenResp> run(BuildContext context, UserCredentials userCredentials) async {
    UserTokenResp result;

    try {
      result = await RpcService.instance.userLogin(userCredentials);
    }
    on RpcException catch(exception) {
      await ShowModalMessage().run(context, Messages.ErrorServer, exception.errors.map((e) => e.errorText).join('\r\n'));
    }
    on TimeoutException {
      await ShowModalMessage().run(context, Messages.ErrorServer, Messages.ErrorTimeout);
    }
    on NoSuchMethodError {
      await ShowModalMessage().run(context, Messages.ErrorUnknown, Messages.ErrorDataStructure);
    }
    catch (exception) {
      await ShowModalMessage().run(context, Messages.ErrorUnknown, Messages.ErrorUserLogin);
    }

    return result;
  }
}
